### ISSUE TITLE
Risk of `form` submission intercept

### ISSUE THREAD

**Issue by willscripted** [ProdPerfect/prodperfect-keen-tracking.js#31]
Needs more eyes:

If a call to `recordEvent` returns false, false may be returned by the `onsubmit` handler of browser autotrack. This may stop the default behavior of form-submission.

The behavior I would expect: even if ProdPerfect fails to collect the form event, the client's code continues to execute uninterrupted.

https://github.com/ProdPerfect/prodperfect-keen-tracking.js/blob/master/lib/browser-auto-tracking.js#L252

**Comment by chanson** [comment]
Slowly piecing through this, jotting down things as i find them

> Of the ~9 or so calls recorded by the fetch mock, the ?arbitrarily chosen? first record is checked

I believe they're just using the first record to check options that are essentially consistent across all different events. They aren't specific details related to form submissions, but it still leaves a confusing test.

The swallowed jsdom error appears to be a known..issue? with jest, the test runner. It just sorta ignores those errors, which ignited a bit of debate on the lib https://github.com/facebook/jest/issues/5266

Despite the jsdom error, the test does appear to test that `form_submission` events get set, via https://github.com/keen/keen-tracking.js/blob/master/test/unit/modules/browser-auto-tracking-spec.js#L331. To make sure I understand what functionaliy we're looking for, we want to make sure that if that event recording fails, the form submission still goes through, which is not currently possible as-is because js-dom does not submit the form? So the proposal is to actually confirm a submission via testcafe.

**Comment by willscripted** [comment]
> So the proposal is to actually confirm a submission via testcafe.

Yes. Start with a base case, and if you extra bandwidth, add a case where an error is raised in the tracking code.
